src/VisualStudio/CSharp/Test/Options/AutomationObjectTests.cs

```diff
@@ -0,0 +1,289 @@
+                        else
+                        {
+                            radioButtonGroups.Add(radioButton.GroupName, new List<RadioButton> { radioButton });
+                        }
+
+                        continue;
+                    }
+                }
+
+                foreach (var radioButtonGroup in radioButtonGroups)
+                {
+                    var groupName = radioButtonGroup.Key;
+                    var radioButtons = radioButtonGroup.Value;
+                    // There is no point in having a single radio button in a group.
+                    Assert.True(radioButtons.Count > 1, $"Expected radio button group '{groupName}' to have more than one radio button. Found {radioButtons.Count}.");
+                    var selectedRadioButton = radioButtons.SingleOrDefault(r => r.IsChecked == true);
+                    foreach (var radioButton in radioButtons)
+                    {
+                        // We test selecting every radio button in the group.
+                        // We skip the already selected one till we are sure we tested other radio buttons.
+                        if (radioButton == selectedRadioButton)
+                        {
+                            continue;
+                        }
+
+                        Assert.False(radioButton.IsChecked);
+                        VerifySingleChangeWhenOptionChangeInUI(automationObject, () => radioButton.IsChecked = true, optionService, optionStore, optionForAssertMessage: radioButton.Name);
+                    }
+
+                    // TODO: Consider asserting a non-null selectedRadioButton if https://github.com/dotnet/roslyn/issues/62363 is fixed.
+
+                    if (selectedRadioButton is not null)
+                    {
+                        // Now that we tested other radio buttons in the group, the initially selected radio button is now not selected.
+                        Assert.False(selectedRadioButton.IsChecked);
+                        VerifySingleChangeWhenOptionChangeInUI(automationObject, () => selectedRadioButton.IsChecked = true, optionService, optionStore, optionForAssertMessage: selectedRadioButton.Name);
+                    }
+                }
+            }
+
+            // Above checks that all options are in AutomationObjects.
+            // TODO: check that all automation object members are in options.
+        }
+
+        private static void VerifySingleChangeWhenOptionChangeInUI(AutomationObject automationObject, Action changeUIControl, ILegacyWorkspaceOptionService optionService, OptionStore optionStore, string optionForAssertMessage)
+        {
+            var automationValuesBeforeChange = GetAutomationDictionary(automationObject);
+
+            changeUIControl();
+
+            // Following simulates the SaveSettingsToStorage call.
+            // Save the changes that were accumulated in the option store.
+            var oldOptions = new SolutionOptionSet(optionService);
```

Comment:
As a follow up, it'd be good to avoid using `SolutionOptionSet` here (and for `OptionStore` in general) Filed issue https://github.com/dotnet/roslyn/issues/62384 to track.

Reply:
@tmat The last commit (move to integration tests) should have fixed this.